File: rust-axum/src/handlers/health.rs
use axum::{extract::State, http::StatusCode, Json};
use chrono::Utc;
use std::sync::Arc;
use std::time::Instant;

use crate::{db::DbPool, models::HealthResponse};

/// Health check endpoint
#[utoipa::path(
    get,
    path = "/api/health",
    tag = "Health",
    responses(
        (status = 200, description = "Service is healthy", body = HealthResponse)
    )
)]
pub async fn health_check(
    State(pool): State<Arc<DbPool>>,
) -> Result<Json<HealthResponse>, (StatusCode, Json<serde_json::Value>)> {
    let start = Instant::now();
    
    // Test database connection
    let db_status = match pool.get().await {
        Ok(_) => "connected",
        Err(_) => "disconnected",
    };
    
    let response_time = start.elapsed().as_millis();
    
    Ok(Json(HealthResponse {
        status: "healthy".to_string(),
        timestamp: Utc::now(),
        database: db_status.to_string(),
        response_time,
    }))
}

File: golang-gin/handlers/health.go
package handlers

import (
	"database/sql"
	"net/http"
	"time"

	"github.com/FCHEHIDI/API_Internal_Tools_Management/golang-gin/models"
	"github.com/gin-gonic/gin"
)

// HealthCheck godoc
// @Summary Health check endpoint
// @Description Check if the API and database are operational
// @Tags health
// @Produce json
// @Success 200 {object} models.HealthResponse
// @Failure 500 {object} models.ErrorResponse
// @Router /health [get]
func HealthCheck(db *sql.DB) gin.HandlerFunc {
	return func(c *gin.Context) {
		start := time.Now()
		
		// Test database connection
		err := db.Ping()
		dbStatus := "connected"
		if err != nil {
			dbStatus = "disconnected"
		}

		responseTime := time.Since(start).Milliseconds()

		response := models.HealthResponse{
			Status:       "healthy",
			Timestamp:    time.Now().Format(time.RFC3339),
			Database:     dbStatus,
			ResponseTime: responseTime,
		}

		if err != nil {
			response.Status = "unhealthy"
			c.JSON(http.StatusInternalServerError, response)
			return
		}

		c.JSON(http.StatusOK, response)
	}
}
